pkg/cloud/amazon/vault/vault_resources.go

```diff
@@ -405,3 +523,54 @@ func createAccessKey(region string, username string) (*string, *string, error) {
 	log.Logger().Debugf("Created the vault access key successfully")
 	return result.AccessKey.AccessKeyId, result.AccessKey.SecretAccessKey, nil
 }
+
+func trimString(str string, delta int) string {
+	if len(str) > delta {
+		return str[:delta]
+	}
+	return str
+}
+
+func defineStatus(status string, stackEvent *cloudformation.StackEvent) string {
+	parts := strings.Split(status, "|")
+	switch strings.TrimSpace(parts[len(parts)-1]) {
+	case cloudformation.ResourceStatusCreateComplete, cloudformation.ResourceStatusDeleteComplete, cloudformation.ResourceStatusUpdateComplete:
+		return util.ColorInfo(status)
+	case cloudformation.ResourceStatusCreateFailed, cloudformation.ResourceStatusDeleteFailed, cloudformation.ResourceStatusUpdateFailed:
+		return util.ColorError(status)
+	case cloudformation.ChangeSetStatusCreateInProgress, cloudformation.ResourceStatusDeleteInProgress, cloudformation.ResourceStatusUpdateInProgress:
+		return util.ColorWarning(status)
+	default:
+		return util.ColorAnswer(status)
+	}
+}
+
+func runStackEventsReporting(donec <-chan bool, svc *cloudformation.CloudFormation, stackName *string) {
+	lastLen := 0
+	fmt.Println()
+	for {
+		select {
+		case <-donec:
+			return
+		default:
+			output, _ := svc.DescribeStackEvents(&cloudformation.DescribeStackEventsInput{
+				StackName: stackName,
+			})
+
+			for i, j := 0, len(output.StackEvents)-1; i < j; i, j = i+1, j-1 {
+				output.StackEvents[i], output.StackEvents[j] = output.StackEvents[j], output.StackEvents[i]
+			}
+
+			if lastLen > 0 {
+				fmt.Printf("\033[%dA", lastLen)
```

Comment:
Maybe consider using a logger instead of printing directly to stdout.

Reply:
What I'm doing with this is rewriting the console at certain points, so I think it wouldn't work with a logger.